command/config_validate.go

```diff
@@ -0,0 +1,85 @@
+package command
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	multierror "github.com/hashicorp/go-multierror"
+	agent "github.com/hashicorp/nomad/command/agent"
+)
+
+type ConfigValidateCommand struct {
+	Meta
+}
+
+func (c *ConfigValidateCommand) Help() string {
+	helpText := `
+Usage: nomad config validate <config_path> [<config_path...>]
+
+  Performs a thorough sanity test on Nomad configuration files. For each file
+  or directory given, the validate command will attempt to parse the contents
+  just as the "nomad agent" command would, and catch any errors.
+
+  This is useful to do a test of the configuration only, without actually
+  starting the agent. This performs all of the validation the agent would, so
+  this should be given the complete set of configuration files that are going
+  to be loaded by the agent. This command cannot operate on partial
+  configuration fragments since those won't pass the full agent validation.
+
+  Returns 0 if the configuration is valid, or 1 if there are problems.
```

Comment:
Let's tighten up the text here for inclusive language, for grammar, to match some existing `nomad agent -config` language, and to remove some redundant phrasing:

```suggestion
  Perform validation on a set of Nomad configuration files. This is useful
  to test the Nomad configuration without starting the agent.

  Accepts the path to either a single config file or a directory of
  config files to use for configuring the Nomad agent. This option may
  be specified multiple times. If multiple config files are used, the
  values from each will be merged together. During merging, values from
  files found later in the list are merged over values from previously
  parsed files.

  This command cannot operate on partial configuration fragments since
  those won't pass the full agent validation. This command does not
  require an ACL token.

  Returns 0 if the configuration is valid, or 1 if there are problems.
```